### ISSUE TITLE
Derive macro configuration does not support `cfg_attr` that expands to multiple attributes

### ISSUE THREAD

**Issue by Veykril** [rust-lang/rust-analyzer#16818]
https://github.com/rust-lang/rust-analyzer/pull/16789

**Comment by Dessix** [comment]
I believe I may have just run into this with `sea-orm` in its latest version.

Rust-Analyzer is reporting the following error when `rustc` accepts the code when using multi-predicate `cfg_attr`s:
```
Missing macro attribute \`rs_type\`
```

For anyone hitting this on conditional code like the following:
```rust
#[cfg_attr(
  feature = "sea-query",
  derive(EnumIter, DeriveActiveEnum),
  sea_orm(
    db_type = "String(StringLen::N(64))",
    rs_type = "String",
  ),
)]
pub enum MyEnum {
  // ...
}
```
Manually expanding to two separate predicative attributes resolves the issue:
```rust
#[cfg_attr(feature = "sea-query", derive(EnumIter, DeriveActiveEnum))]
#[cfg_attr(feature = "sea-query", sea_orm(rs_type = "String", db_type = "String(StringLen::N(64))"))]
pub enum MyEnum {
  // ...
}
```

I am unsure why this started cropping up only in the most recent version of SeaORM or rust-analyzer for me, but [the Rust reference documentation](https://doc.rust-lang.org/reference/conditional-compilation.html#conditional-compilation) shows that this attribute accepts multiple items in its `ConfigurationPredicateList` grammar rule. I hope this is what this issue was about, but the linked thread didn't go into much detail about what "handl[ing] multiple attributes" entailed. I'm posting this for searchability, as I spent a while digging through sea-orm instead of RA trying to track this one down.

**Comment by Veykril** [comment]
Ah, you are running into something differently here. I believe the case where `cfg_attr` expands to multiple things is fixed already (what this issue was about). What you are running into is a very special case that we haven't considered, namely where the `derive` is within a `cfg_attr` that expands to more things. When we expand the derive, we have to strip the `derive` invocation before passing the item as the attribute, but we do that pre-cfg_attr expansion so we strip the entire `cfg_attr` in your first example and which is why the second snippet works fine.

**Comment by Veykril** [comment]
https://github.com/rust-lang/rust-analyzer/issues/8434 has some more context

**Comment by Dessix** [comment]
My apologies for posting it to the wrong issue then- but I'm glad I helped catch a subtle bug ^^

**Comment by wyatt-herkamp** [comment]
@Veykril and @Dessix 

This issue already exists here. https://github.com/rust-lang/rust-analyzer/issues/10110

I meant to fix it. However, I have not had time recently.
